src/Admin/BaseFieldDescription.php

```diff
@@ -137,14 +137,15 @@ abstract class BaseFieldDescription implements FieldDescriptionInterface
     private static $fieldGetters = [];
 
     /**
-     * NEXT_MAJOR: Remove the null default value and restrict param type to `string`.
+     * NEXT_MAJOR: Remove the null default value for $name and restrict param type to `string`.
      */
     public function __construct(
         ?string $name = null,
         array $options = [],
         array $fieldMapping = [],
         array $associationMapping = [],
-        array $parentAssociationMappings = []
+        array $parentAssociationMappings = [],
+        ?string $fieldName = null
```

Comment:
I'm okay with adding it as a constructor parameter, but I'm note sure if it should be the last parameter.

We should either use static constructors or use a builder pattern here.

Reply:
Can you explain what you mean by builder pattern or static constructor ?

It’s already the ModelManager job to build the FieldDescription in the method « getNewFieldDescriptionInstance ».

And for the fact the fieldName is last:
- It ensure to be BC
- The fieldName is sometimes facultative.

When there is no parent association mapping, the fieldname will not be needed in next major (since the fieldname will be equal to the name).

Reply:
> Can you explain what you mean by builder pattern or static constructor ?

Sure.

The static constructor pattern is something that doctrine is doing for some time. This allows creating two or more named constructors for a class:
https://github.com/doctrine/orm/blob/3ef5a30102905483c29a6849c8c4efb4dfbe97d6/lib/Doctrine/ORM/Query/QueryException.php#L252-L256

The builder pattern is explained here: https://refactoring.guru/design-patterns/builder/php/example. We already use it for the [doctrine-extensions](https://github.com/sonata-project/sonata-doctrine-extensions/blob/1.x/src/Mapper/Builder/OptionsBuilder.php). It is normally used to construct new objects or configurations. The basic idea is to chain the different calls.

Applying this pattern to the field description requires some more refactoring but can reduce the dependencies that are added via the constructor or setters.

Reply:
I do like the FieldDescriptionBuilder, will create an issue for this.